tests/dateSelector.spec.ts

```diff
@@ -0,0 +1,90 @@
+        await expect(page.getByRole("heading")).toHaveText("New Visit")
+        await expect(page.locator('tr', { hasText: 'Samantha' })).toContainText('Jean Coleman')
+        await page.getByLabel('Open calendar').click()
+        await page.locator('span.mat-calendar-body-today').click()
+        const date = new Date()
+        const year = date.getFullYear();
+        const month = date.toLocaleString('en-US', { month: '2-digit' })
+        const day = date.toLocaleString('en-US', { day: '2-digit' })
+        const expectedDermatologyAppointmentDate = `${year}-${month}-${day}`
+        await expect(page.locator('.mat-datepicker-input')).toHaveValue(`${year}/${month}/${day}`)
+        await page.locator('#description').fill('dermatology visit')
+        await page.getByRole('button', { name: 'Add Visit' }).click()
+        await expect(petsAndVisitsRowSamatha.locator('app-visit-list tr td').first()).toHaveText(expectedDermatologyAppointmentDate)
+        await petsAndVisitsRowSamatha.getByText('Add Visit').click()
+        await page.getByLabel('Open calendar').click()
+        date.setDate(date.getDate() - 45)
+        const previousAppointmentYear = date.getFullYear();
+        const previousAppointmentMonth = date.toLocaleString('en-US', { month: '2-digit' })
+        const previousAppointmentDayNumeric = date.toLocaleString('en-US', { day: 'numeric' })
+        const previousAppointmentDay2Digit = date.toLocaleString('en-US', { day: '2-digit' })
+        const expectedMassageAppointmentDate = `${previousAppointmentYear}-${previousAppointmentMonth}-${previousAppointmentDay2Digit}`;
+        let calendarMonthAndYear = await page.locator('.mat-calendar-period-button').textContent()
+        while (!calendarMonthAndYear?.includes(`${previousAppointmentMonth} ${previousAppointmentYear}`)) {
+            await page.locator('.mat-calendar-previous-button').click()
+            calendarMonthAndYear = await page.locator('.mat-calendar-period-button').textContent()
+        }
+        await page.getByText(previousAppointmentDayNumeric, { exact: true }).click()
+        await page.locator('#description').fill('massage therapy')
+        await page.getByText('Add Visit').click()
+        const petAllAppointments = petsAndVisitsRowSamatha.locator('app-visit-list tr')
+        const laterAppointmentDateforSamathasPet = await petAllAppointments.nth(1).locator('td').first().textContent()
+        const currentAppointmentDateforSamathasPet = await petAllAppointments.nth(2).locator('td').first().textContent()
+        expect(Date.parse(currentAppointmentDateforSamathasPet!) < Date.parse(laterAppointmentDateforSamathasPet!)).toBeTruthy()
+        await petsAndVisitsRowSamatha.locator('app-visit-list tr', { hasText: expectedDermatologyAppointmentDate }).getByText('Delete Visit').click()
+        await petsAndVisitsRowSamatha.locator('app-visit-list tr', { hasText: expectedMassageAppointmentDate }).getByText('Delete Visit').click()
```

Comment:
Since you have create the locator for pet appointments in line 76, you could reuse this locator here and use filter() method.
```
await petAllAppointments.filter({hasText: expectedDermatologyAppointmentDate}).getByText('Delete Visit').click()
await petAllAppointments.filter({hasText: expectedMassageAppointmentDate}).getByText('Delete Visit').click()
```

It makes code more compact and reduce copy/pasting